### PR TITLE
En 891 user aware api and bugs fix

### PR DESCRIPTION
**Pull request by paddeo**

### CODE REVIEW

**Comment by entando-service** [comment]
All is well

File: engine/src/main/java/org/entando/entando/aps/system/common/model/DeleteDto.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/common/model/DeleteDto.java]: Need to fix this header.

File: engine/src/main/java/org/entando/entando/aps/system/common/model/DeleteDto.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+package org.entando.entando.aps.system.common.model;
+
+/**
+ *
+ * @author paddeo
+ */
+public class DeleteDto {
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/common/model/DeleteDto.java]: I don't think we want to do this. Generally, not a good thing to have objects that represent verbs. If we need to return a spares object either return a Map with the few things in it you need or a sparsely populated instance of the thing being deleted.

File: engine/src/main/java/org/entando/entando/web/page/PageController.java [hunk 3]
```diff
@@ -138,12 +175,17 @@ public void setPageValidator(PageValidator pageValidator) {
             throw new ValidationGenericException(bindingResult);
         }
         this.getPageService().removePage(pageCode);
-        return new ResponseEntity<>(new RestResponse(pageCode), HttpStatus.OK);
+        DeleteDto dto = new DeleteDto(pageCode);
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/web/page/PageController.java]: see comment above. Refactor this class out and either return a sparse Page or return a Map

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by paddeo** [comment]
retest this please

**Comment by entando-service** [comment]
All is well

File: engine/src/main/java/org/entando/entando/aps/system/services/auth/IAuthorizationService.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/auth/IAuthorizationService.java]: Fix this header in your next PR

File: engine/src/main/java/org/entando/entando/aps/system/services/page/IPageService.java [hunk 5]
```diff
@@ -1,47 +1,47 @@
...
-    public WidgetConfigurationDto updateWidgetConfiguration(String pageCode, int frameId, WidgetConfigurationRequest widget);
-
-    public void deleteWidgetConfiguration(String pageCode, int frameId);
-
-}
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/page/IPageService.java]: Fix in the next PR

File: engine/src/main/java/org/entando/entando/aps/system/services/page/PageAuthorizationService.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/page/PageAuthorizationService.java]: Fix in next PR

File: engine/src/main/java/org/entando/entando/web/page/PageController.java [hunk 7]
```diff
@@ -71,23 +83,42 @@ public void setPageValidator(PageValidator pageValidator) {
...
+    }
+
+    public void setAuthorizationService(PageAuthorizationService authorizationService) {
+        this.authorizationService = authorizationService;
+    }
+
+    @RestAccessControl(permission = Permission.MANAGE_PAGES)
+    @RequestMapping(value = "/pages", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> getPages(@ModelAttribute("user") UserDetails user, @RequestParam(value = "parentCode", required = false, defaultValue = "homepage") String parentCode) {
+        List<PageDto> result = this.getAuthorizationService().filterList(user, this.getPageService().getPages(parentCode));
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/web/page/PageController.java]: Add debug logging to controller methods

File: engine/src/main/java/org/entando/entando/web/page/PageController.java [hunk 8]
```diff
@@ -71,23 +83,42 @@ public void setPageValidator(PageValidator pageValidator) {
...
+        metadata.put("parentCode", parentCode);
+        return new ResponseEntity<>(new RestResponse(result, new ArrayList<>(), metadata), HttpStatus.OK);
     }
 
-    @RestAccessControl(permission = Permission.SUPERUSER)
-    @RequestMapping(value = "/page/{pageCode}", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
-    public ResponseEntity<?> getPage(@PathVariable String pageCode, @RequestParam(value = "status", required = false, defaultValue = IPageService.STATUS_DRAFT) String status) {
+    @RestAccessControl(permission = Permission.MANAGE_PAGES)
+    @RequestMapping(value = "/pages/{pageCode}", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> getPage(@ModelAttribute("user") UserDetails user, @PathVariable String pageCode, @RequestParam(value = "status", required = false, defaultValue = IPageService.STATUS_DRAFT) String status) {
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/web/page/PageController.java]: Add debug logging to controller methods

File: engine/src/main/java/org/entando/entando/web/page/PageController.java [hunk 9]
```diff
@@ -98,12 +129,13 @@ public void setPageValidator(PageValidator pageValidator) {
...
-        return new ResponseEntity<>(new RestResponse(page), HttpStatus.OK);
+        Map<String, String> metadata = new HashMap<>();
+        return new ResponseEntity<>(new RestResponse(page, new ArrayList<>(), metadata), HttpStatus.OK);
     }
 
-    @RestAccessControl(permission = Permission.SUPERUSER)
+    @RestAccessControl(permission = Permission.MANAGE_PAGES)
     @RequestMapping(value = "/pages", method = RequestMethod.POST, produces = MediaType.APPLICATION_JSON_VALUE)
-    public ResponseEntity<?> addPage(@Valid @RequestBody PageRequest pageRequest, BindingResult bindingResult) throws ApsSystemException {
+    public ResponseEntity<?> addPage(@ModelAttribute("user") UserDetails user, @Valid @RequestBody PageRequest pageRequest, BindingResult bindingResult) throws ApsSystemException {
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/web/page/PageController.java]: Add debug logging to controller methods with body

File: engine/src/main/java/org/entando/entando/web/page/PageController.java [hunk 10]
```diff
@@ -114,13 +146,17 @@ public void setPageValidator(PageValidator pageValidator) {
...
-        return new ResponseEntity<>(new RestResponse(dto), HttpStatus.OK);
+        Map<String, String> metadata = new HashMap<>();
+        return new ResponseEntity<>(new RestResponse(dto, new ArrayList<>(), metadata), HttpStatus.OK);
     }
 
-    @RestAccessControl(permission = "page_delete")
+    @RestAccessControl(permission = Permission.MANAGE_PAGES)
     @RequestMapping(value = "/pages/{pageCode}", method = RequestMethod.DELETE, produces = MediaType.APPLICATION_JSON_VALUE)
-    public ResponseEntity<?> deletePage(@PathVariable String pageCode) throws ApsSystemException {
+    public ResponseEntity<?> deletePage(@ModelAttribute("user") UserDetails user, @PathVariable String pageCode) throws ApsSystemException {
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/web/page/PageController.java]: Add debug logging to controller methods

**Pull request merged**